### PR TITLE
AWS S3 - Add aws-s3-get-bucket-encryption command

### PR DESCRIPTION
**Pull request by icholy**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description

Adds a `aws-s3-get-bucket-encryption` command to the AWS S3 integration.

## Screenshots
image

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/AWS-S3/Integrations/AWS-S3/AWS-S3.yml [hunk 1]
```diff
@@ -619,6 +619,32 @@ script:
...
+    - contextPath: AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.ApplyServerSideEncryptionByDefault.SSEAlgorithm
+      description: S3 Bucket Encryption SSE Algorithm.
+      type: string
+    - contextPath: AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.ApplyServerSideEncryptionByDefault.KMSMasterKeyID
+      description: S3 Bucket Encryption KMS Master Key ID.
+      type: string
+    - contextPath: AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.BucketKeyEnabled
+      description: S3 Bucket Encryption Key Enabled.
+      type: boolean
+    arguments:
```
Comment (YuvHayun) [inline Packs/AWS-S3/Integrations/AWS-S3/AWS-S3.yml:634]: let's move the arguments to the top (line 622)

File: Packs/AWS-S3/Integrations/AWS-S3/README.md [hunk 2]
```diff
@@ -419,3 +419,38 @@ There is no context output for this command.
...
+
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.ApplyServerSideEncryptionByDefault.SSEAlgorithm | String | S3 Bucket Encryption SSE Algorithm. |
+| AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.ApplyServerSideEncryptionByDefault.KMSMasterKeyID | String | S3 Bucket Encryption KMS Master Key ID. |
+| AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.BucketKeyEnabled | Boolean | S3 Bucket Encryption Key Enabled. |
+
+#### Command Example
+``` !aws-s3-put-public-access-block bucket="bucket name" BlockPublicAcls=True IgnorePublicAcls=False BlockPublicPolicy=True RestrictPublicBuckets=True```
+
```
Comment (YuvHayun) [inline Packs/AWS-S3/Integrations/AWS-S3/README.md:453]: Let's also add the '#### Context example' section here

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hi @icholy, thx for your contributions.
Please see my comments below.
Also:
1. Please add unit tests to the command you just added (you'll need to create `AWS-SE_test.py` file).
2. Please add a step to the test playbook with your command.
3. Please add release notes. To do so, enter `demisto-sdk update-release-notes -i  Packs/AWS-S3` and replace the "%%UPDATE_RN%%" line with your additions. (enter `demisto-sdk doc-review --templates   ` to list the optional templates).


Please don't hesitate to contact me with any questions, I'm available both here and via slack.
Thanks again!

**Comment by icholy** [comment]
@YuvHayun thanks for the prompt review! Can you point me toward some existing tests I can use as an example?

**Comment by YuvHayun** [comment]
> @YuvHayun thanks for the prompt review! Can you point me toward some existing tests I can use as an example?

@icholy sure, I'd recommend 'HEllowWorld_test.py' as an example.
In general, the whole pack is an example pack of how a pack should look like.

**Comment by YuvHayun** [comment]
@icholy 
Seems like there're a lot of problems due to the PR being untouched for too long.
Please merge for master and solve conflicts.

**Comment by icholy** [comment]
Oh cool, you guys updated it!

File: Packs/AWS-S3/Integrations/AWS-S3/AWS-S3.yml [hunk 3]
```diff
@@ -505,6 +505,32 @@ script:
...
+    description: Get AWS S3 Bucket Encryption
+    name: aws-s3-get-bucket-encryption
+    outputs:
+    - contextPath: AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.ApplyServerSideEncryptionByDefault.SSEAlgorithm
+      description: S3 Bucket Encryption SSE Algorithm.
+      type: string
+    - contextPath: AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.ApplyServerSideEncryptionByDefault.KMSMasterKeyID
+      description: S3 Bucket Encryption KMS Master Key ID.
+      type: string
+    - contextPath: AWS.S3.Buckets.BucketName.ServerSideEncryptionConfiguration.Rules.BucketKeyEnabled
```
Comment (icholy) [inline Packs/AWS-S3/Integrations/AWS-S3/AWS-S3.yml:531]: @YuvHayun I can't tell if I'm setting the `contextPath` correctly. Do I need that `BucketName` property in there? The `aws-s3-get-public-access-block` command includes the `BucketName` in the path, but the rest of the commands do not.

Comment (YuvHayun) [reply Packs/AWS-S3/Integrations/AWS-S3/AWS-S3.yml:531]: Yes, keep the bucket name please.

File: Packs/AWS-S3/ReleaseNotes/1_2_3.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### AWS - S3
+- Added the **aws-s3-get-bucket-encryption** command.
```
Comment (YuvHayun) [inline Packs/AWS-S3/ReleaseNotes/1_2_3.md:4]:
```suggestion
- Added the **aws-s3-get-bucket-encryption** command.
- Upgraded the Docker image to *demisto/boto3py3:1.0.0.34229*.
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
@icholy The fixes look great!
I added 1 more comment.
Also, could you please add this command to the Test playbook and add a quick video of you running the command and showing the outputs you get and everything?

**Comment by YuvHayun** [comment]
@icholy Thx for the fix.
Please add this command to the Test playbook and add a quick video of you running the command and showing the outputs you get.

**Comment by icholy** [comment]
https://user-images.githubusercontent.com/943597/191582563-93bd50e7-5520-4b52-b953-c797e25dc4ea.mp4

**Comment by YuvHayun** [comment]
@icholy all validations passed, thx for your contribution!
merging this pr.

**Review by YuvHayun — APPROVED** [review]

**Pull request merged**